NethServer/ns8-core
# Firewalld integration

Use firewalld to setup a basic firewall.
The firewall has 2 zones:
- trusted: it includes containers and wireguard interface
- public: everything else

To test it, use:
```
curl https://raw.githubusercontent.com/NethServer/ns8-core/firewalld/core/install.sh > install.sh
bash install.sh ghcr.io/nethserver/core:firewalld
```
Podman firewalld integration is only partial:
- podman 4, in CentOS Stream, automatically configure itself for firewalld using netavark
- pdoman 3.x, on Debian and Ubuntu, needs to explicitly configure CNI plugin
Please note that on both scenarios, rootfull containers with DNAT are not supported.

The `firewall-setup` script:
- takes care of CNI plugin configuration
- setup default policies
- makes sure SSH port is open
- enable logging of blocked packets 

Refactored implementation adds `node/add-public-service` and `node/remove-public-service` APIs.
Modules need the `org.nethserver.authorizations=node:fwadm` label to use such APIs.

Example of a step inside the create-module:
```python
import os
import agent
agent.assert_exp(agent.add_public_service(os.environ['MODULE_ID'], ["80/tcp", "443/tcp"]))
```

Example of a step inside the remove-module:
```python
import os
import agent
agent.assert_exp(agent.remove_public_service(os.environ['MODULE_ID']))
```

Current implementation has the following problem: traefik can't use the `add-public-service` API because the underlying function needs traefik up&running, so the NethServer/ns8-traefik#5 has been closed because it doesn't work at all.
